service/smithyprototype/lexruntimeservice/deserialize.go

```diff
@@ -0,0 +1,1444 @@
+package lexruntimeservice
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/internal/sdkio"
+	"github.com/aws/aws-sdk-go-v2/service/smithyprototype/lexruntimeservice/types"
+	"github.com/awslabs/smithy-go/middleware"
+	smithyHTTP "github.com/awslabs/smithy-go/transport/http"
+)
+
+type awsRestJson1_deserializeOpPutSession struct{}
+
+// ID is the middleware identifier
+func (p awsRestJson1_deserializeOpPutSession) ID() string {
+	return "awsRestJson1_deserializeOpPutSession"
+}
+
+func (p awsRestJson1_deserializeOpPutSession) HandleDeserialize(ctx context.Context, in middleware.DeserializeInput, next middleware.DeserializeHandler) (
+	out middleware.DeserializeOutput, metadata middleware.Metadata, err error,
+) {
+	out, metadata, err = next.HandleDeserialize(ctx, in)
+	if err != nil {
+		return out, metadata, err
+	}
+	response, ok := out.RawResponse.(*smithyHTTP.Response)
+	if !ok {
+		return out, metadata, &aws.DeserializationError{
+			Err: fmt.Errorf("unknown transport type %T", out.RawResponse),
+		}
+	}
+
+	output, ok := out.Result.(*PutSessionOutput)
+	if !ok {
+		return out, metadata, &aws.DeserializationError{
+			Err: fmt.Errorf("unknown output result type %T", out.Result),
+		}
+	}
```

Comment:
We'll want to consider which middleware will populate `out.Result` with the operation's output type. The operation serialize probably is a good opportunity for that. Alternatively a separate deserialize middleware that does nothing more than set the `out.Result` member to an empty op output type value.

Reply:
out.Result is an `interface{}`and here we are just type casting it to `*PutSessionOutput` before passing it to protocol deserializer funcs.

If you were talking about populating `out.RawResponse`, then I think a seperate deserialize middleware or a client middleware that actually takes in the serialized request, send it to server and returns a response can handle that.

Reply:
Cool in that case, yes a separate middleware added by the operation building the stack would need to take care of populating the Result member with a empty output type value.

Reply:
I don't see a reason to keep the middleware that set's the `out.Result` separate. If in case some customization needs it in the future, we can add the middleware, we can change this. 

I will go ahead and set the type in this middleware, instead of trying to type assert it.

Reply:
Updated the deserializer middleware to initialize the `out.Result` with output type value. This did not need to be a separate middleware. Initializing with an empty output type value ensures that we overwrite any existing value in `out.Result`